### PR TITLE
Refactor notifications: remove async task

### PR DESCRIPTION
**Pull request by mkevins**
### Description

This PR removes the usage of `AsyncTask` in `NotificationsDetailListFragment`. This is an alternative approach to extracting a use-case class, which has a considerably less complex change set.

Note: this is built on top of https://github.com/wordpress-mobile/WordPress-Android/pull/20467 and should be merged _after_ that PR.

<details>
<summary>Expand for more details on the use-case approach</summary>

The attempt can be found on [this branch](https://github.com/wordpress-mobile/WordPress-Android/compare/experimental/hack-week-2024-03-11/refactor-notifications--viewpager2...experimental/hack-week-2024-03-11/refactor-notifications--remove-async-task--dead-end-extraction) (not to be merged). Specifically, this commit: https://github.com/wordpress-mobile/WordPress-Android/commit/b2209852d40827b7771f855d6b5b4452023bcd40

There is a high degree of coupling with the methods used within the inner AsyncTask class, as well as many reference to the enclosing scope. Some of these can be mitigated by expanding the arity of the method signatures, but in many cases this was already strained (a code smell). The deal-breaker, imo, is that there aren't only stateful references, but direct side-effects leaking into the outer class. This makes extraction very cumbersome because in order to do so cleanly, several new mechanisms would need to be implemented to manage these state changes in both directions. The complexity cost of this would be likely be greater than leaving it as it is in the deprecated state.

Probably, in the future, it would be best to attempt extracting these methods individually, once a view model is established to house the state logic, and especially after the data has been modeled with immutable types.

</details>

tl;dr this approach favors a smaller set of changes, keeping most of the coupled logic in the fragment class. Instead of an inner AsyncTask class, an inner object is created to house the same methods. Methods which are accessing disk or network as converted to suspended methods, and coroutine dispatchers are used to put the work on the appropriate threads.

-----

## To Test:

Smoke test (behavior should not change):

* Log in to an account with notifications
* Tap the notifications tab
* Tap one of the notifications (expect to see the detail view)
* Swipe left and right to view other notifications in the list
* Expect the transitions to be working as before

-----

## Regression Notes

1. Potential unintended areas of impact

Notifications Detail View

2. What I did to test those areas of impact (or what existing automated tests I relied on)

Manual testing

3. What automated tests I added (or what prevented me from doing so)

The code is not yet in a state to facilitate adding unit tests for these changes.

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):

- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by antonis — APPROVED** [review]
Thank you for this improvement @mkevins 🙇 

> tl;dr this approach favors a smaller set of changes, keeping most of the coupled logic in the fragment class. Instead of an inner AsyncTask class, an inner object is created to house the same methods. Methods which are accessing disk or network as converted to suspended methods, and coroutine dispatchers are used to put the work on the appropriate threads.

The approach chosen makes sense as a first step on improving the code 👍 
I've updated the branch with the head of trunk and tested the feature on a Pixel 8 with Android 14:
✅ Notifications fetch on a fresh install
✅ List refresh
✅ Receiving and tapping a new notification
✅ Browsing the notification details for various notification types
✅ Swiping between notifications
I didn't encounter any breakages and the code looks good and reduces the code debt 🎉

**Comment by mkevins** [comment]
Thank you for reviewing this, and testing it out with all those scenarios! Much appreciated!
